### ISSUE TITLE
[BUG] - Byron address attributes required on MC4

### ISSUE THREAD

**Issue by SebastienGllmt** [input-output-hk/cardano-node#1578]
**Internal**

**Summary**

Mainnet candidates are supposed to mimic mainnet. One of the behaviors for mainnet is that the protocol magic is NOT required inside Byron address attributes (as opposed to a testnet where it IS required)

However, MC4 contains the two following issues:
1) Cardano-CLI signing using a bootstrap witness adds the protocol magic to the witness
2) Cardano-Node will reject any transaction where the bootstrap witness does NOT contain the protocol magic

**Steps to reproduce**

Key file (`payment_byron.skey`)
```
{
    "type": "PaymentSigningKeyByron_ed25519_bip32",
    "description": "Payment Signing Key",
    "cborHex": "5880b8f2bece9bdfe2b0282f5bad705562ac996efb6af96b648f4445ec44f47ad95c10e3d72f26ed075422a36ed8585c745a0e1150bcceba2357d058636991f38a37cf76399a210de8720e9fa894e45e41e29ab525e30bc402801c076250d1585bcd91e248de509c070d812ab2fda57860ac876bc489192c1ef4ce253c197ee219a4"
}
```
Notably, this corresponds to the address `Ae2tdPwUPEZEY1CHq8oQxLm523fWkN7V7AK2WKiQTi8thiA9qZJpsEQgn95`

Transaction file (`tx.raw`)
```
{
    "type": "TxUnsignedShelley",
    "description": "",
    "cborHex": "82a4008182582062c337c54795149010a805ee42eb8554d8db5c327d9d88e0b83529fe42fde57b00018182581d614375a6c397e9c62bf574b9f37bbe25ff4e4a1ac09982117f296c7a521a00895440021a000f4240031a00037460f6"
}
```

Command
```
cardano-cli shelley transaction sign --tx-body-file tx.raw --signing-key-file payment_byron.skey --testnet-magic 42 --out-file tx.signed
```

**Expected behavior**

1) Bootstrap witness on a mainnet candidate should NOT require a protocol magic to be accepted by the node
2) Bootstrap witness generated by cardano-cli should NOT contain a protocol magic

**System info (please complete the following information):**
- 1.18.0

**Screenshots and attachments**

You can see from the CBOR that the Cardano-CLI generated transaction contains the prtocol magic "42" inside it (whereas it's expected not to contain a protocol magic at all)
![image](https://user-images.githubusercontent.com/2608559/88650753-f7292f80-d103-11ea-8b18-22df57fc0325.png)

**Additional context**

https://github.com/input-output-hk/cardano-node/blob/master/cardano-api/src/Cardano/Api/Typed.hs#L1325-L1333
```
Matthias B.  22 hours ago
This seems wrong as it'll insert a protocol magic attribute if an explicit magic is given to the command-line. We had a similar issue in cardano-wallet (treating staging networks as if they were testnets) and resorted to having three modes:
--mainnet
--testnet MAGIC
--staging MAGIC
where --staging implies RequiresNoMagic
```

This is the transaction that should be accepted
```
83a4008182582062c337c54795149010a805ee42eb8554d8db5c327d9d88e0b83529fe42fde57b00018182581d614375a6c397e9c62bf574b9f37bbe25ff4e4a1ac09982117f296c7a521a00895440021a000f4240031a00037460a102818458208621b4ed226c33282bea86f495aa589349280f739032f9cdb847e393a6bd32e7584071f410a3a4d684c4a0dcdc451b1a647a6fd29f708e10d879e5770ebeea4b25c76e493628faa0bb03f90cb8bb2d9fdf284158bc64d91a00f79f2705669f65160e5820f1720f17e7573e467a1a582b0b0c4d84d8c638448168996310f69341c94b87e141a0f6
```
Created with the following steps

```
cardano-cli shelley transaction build-raw \
--tx-in 62c337c54795149010a805ee42eb8554d8db5c327d9d88e0b83529fe42fde57b#0 \
--tx-out Ae2tdPwUPEZEY1CHq8oQxLm523fWkN7V7AK2WKiQTi8thiA9qZJpsEQgn95+9000000 \
--ttl 226400 \
--fee 1000000 \
--out-file tx.raw

cardano-cli shelley transaction sign --tx-body-file tx.raw --signing-key-file payment_byron.skey --mainnet --out-file tx.signed
```

**Comment by dcoutts** [comment]
Use `--mainnet` for the signing step.

**Comment by SebastienGllmt** [comment]
I've been told this was fixed in Cardano-Rest 2.1.1 (see changelogs [here](https://github.com/input-output-hk/cardano-rest/releases/tag/2.1.1))
